packages/lesswrong/server/sql/SelectQuery.ts

```diff
@@ -343,9 +343,14 @@ class SelectQuery<T extends DbObject> extends Query<T> {
       this.atoms.push("ORDER BY");
       const sorts: string[] = [];
       for (const field in sort) {
-        const pgSorting = sort[field] === 1
-            ? "ASC NULLS FIRST"
-            : "DESC NULLS LAST"
+        const fieldType = this.table.getField(field)
+        const fieldIsNonnull =
+          (fieldType instanceof NotNullType)
+          || (fieldType instanceof DefaultValueType && fieldType.isNotNull());
```

Comment:
I think this is indicative of `DefaultValueType` having a bug that's probably worth fixing as it may be causing issues elsewhere. It has:
```
  isNotNull(): boolean {
    return this.type instanceof NotNullType;
  }
```
which should almost certainly be:
```
  isNotNull(): boolean {
    return this.type.isNotNull();
  }
```